### PR TITLE
Ignore cycles

### PR DESCRIPTION
**Pull request by jmafoster1**
Added an `--ignore_cycles` option to `main_dafni.py` to bypass the DAG validity check for the EA case study

### CODE REVIEW

File: causal_testing/specification/causal_dag.py [hunk 1]
```diff
@@ -164,16 +169,14 @@ def check_iv_assumptions(self, treatment, outcome, instrument) -> bool:
...
-                cause
-                for cause in self.graph.nodes
-                if list(nx.all_simple_paths(self.graph, source=cause, target=instrument))
-                and list(nx.all_simple_paths(self.graph, source=cause, target=outcome))
-            )
-        ):
-            raise ValueError(f"Instrument {instrument} and outcome {outcome} share common causes")
 
+        for cause in self.graph.nodes:
+            if cause not in (instrument, outcome):  # exclude self-cycles due to breaking changes in NetworkX > 3.2
```
[the remark this answers is not in the document]
Comment (f-allian) [reply causal_testing/specification/causal_dag.py:174]: @jmafoster1 Can you explain what you mean by this, please? This syntax is equivalent to:

``` if cause != instrument and cause != outcome``` which is thought we wanted.

Do you mean there are instances where we need to check the nodes individually?

Also answering your two other comments here. Are you after something like this?

```for cause in self.graph.nodes:
            if cause not in instrument:  
                outcome_paths = list(nx.all_simple_paths(self.graph, source=cause, target=outcome))
             if cause not in outcome:
                 instrument_paths = list(nx.all_simple_paths(self.graph, source=cause, target=instrument))
                ....  ```

[the remark this answers is not in the document]
Comment (jmafoster1) [reply causal_testing/specification/causal_dag.py:174]: Yes, we need two tests like you have above. I was envisaging something like
```
outcome_paths = list(nx.all_simple_paths(self.graph, source=cause, target=outcome)) if cause != outcome else []
instrument_paths = list(nx.all_simple_paths(self.graph, source=cause, target=instrument)) if cause != instrument else []
```
That should give us the exact same results as we were getting from networkx before.

**Comment by f-allian** [comment]
@jmafoster1 Looks like there are also some codecov patches that are failing, could you have a look at them when you have time?

**Review by f-allian — APPROVED** [review]
@jmafoster1 LGTM

**Pull request merged**